src/jit/decomposelongs.cpp

```diff
@@ -947,6 +952,7 @@ GenTree* DecomposeLongs::DecomposeArith(LIR::Use& use)
     loResult->gtOp.gtOp2 = loOp2;
 
     GenTree* hiResult = new (m_compiler, oper) GenTreeOp(GetHiOper(oper), TYP_INT, hiOp1, hiOp2);
+    hiResult->gtFlags |= tree->gtFlags & GTF_ALL_EFFECT;
```

Comment:
The only relevant side effect is likely `GTF_EXCEPT`; the others should be omitted.